### PR TITLE
Implement SIMD bools in trampoline

### PR DESCRIPTION
**Pull request by afonso360**
This PR adds support for SIMD bools in the trampoline.

Its based on #3304, we probably should wait until that one is merged before reviewing this.  So that changes don't appear here as well.

Alternatively I can push these changes onto that PR, but i felt that it would be too many changes for a single PR.

Fixing trampoline bool args is tracked in #2237

### CODE REVIEW

**Comment by abrown** [comment]
@afonso360, can you rebase?

File: cranelift/filetests/src/function_runner.rs [hunk 1]
```diff
@@ -297,13 +297,16 @@ fn make_trampoline(signature: &ir::Signature, isa: &dyn TargetIsa) -> Function {
...
-            let ty = if param.value_type.is_bool() {
-                Type::int(max(param.value_type.bits(), 8)).expect(
-                    "to be able to convert any boolean type to its equal-width integer type",
-                )
+            let is_scalar_bool = param.value_type.is_bool();
+            let is_vector_bool =
+                param.value_type.is_vector() && param.value_type.lane_type().is_bool();
+
+            let ty = if is_scalar_bool || is_vector_bool {
+                param.value_type.as_int()
```
Comment (cfallin) [inline cranelift/filetests/src/function_runner.rs:305]: Can you turn this sequence into a helper on the type itself? Something like `param.value_type.coerce_bools_to_ints()` would be reasonable, IMHO...

**Comment by afonso360** [comment]
Done!

**Review by cfallin — APPROVED** [review]
Thanks! It looks like just the last two comments are new; once rebased (on preview, as @abrown says above) and with the minor comment below addressed, LGTM.

**Comment by afonso360** [comment]
Oops, forgot about this PR, should be ok now.

**Review by cfallin — APPROVED** [review]
Thanks!

**Pull request merged**